tools/train.py

```diff
@@ -115,7 +116,9 @@ def main():
 
     datasets = [build_dataset(cfg.data.train)]
     if len(cfg.workflow) == 2:
-        datasets.append(build_dataset(cfg.data.val))
+        val_dataset = copy.deepcopy(cfg.data.val)
+        val_dataset.pipeline = cfg.train_pipeline
```

Comment:
Change `cfg.train_pipeline` to `cfg.data.train.pipeline` since `cfg.train_pipeline` may not exist.

Reply:
yeah